shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -114,6 +115,14 @@ - (instancetype)initWithName:(NSString*)labelPrefix
   else
     _dartProject.reset([project retain]);
 
+  if (!EnableTracingIfNecessary([_dartProject.get() settings])) {
+    NSLog(@"Cannot create a FlutterEngine instance in debug mode without Flutter tooling or Xcode. "
+          @"To create FlutterEngine instances from applications launched from the home-screen, use "
+          @"profile or release modes instead.");
+    [self release];
```

Comment:
Should we assert here or do something more eye-catching than a console message?

Reply:
If we assert, we will take down the process. We don't want to do that. Instead, we want to display an error message in existing applications.